### PR TITLE
Don't rely on bashisms

### PR DESCRIPTION
**Pull request by fgtham**
Fixes issue #200:

```
$ CONFIG_SHELL=/bin/dash ./configure && make check
[...]
============================================================================
Testsuite summary for rcm 1.3.1
============================================================================
# TOTAL: 36
# PASS:  36
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

### CODE REVIEW

File: bin/rcup.in [hunk 1]
```diff
@@ -136,7 +136,7 @@ is_linked() {
 
   if [ -h "$dest" ]; then
     local link_dest=$(readlink $dest)
-    [ "$link_dest" == "$src" ]
+    expr "$link_dest" == "$src" >/dev/null
```
Comment (mike-burns) [inline bin/rcup.in]: You don't have `[`? Or, I don't understand what's wrong with the `[` program?

**Comment by mike-burns** [comment]
I'd hesitate to call these bashisms (especially since I develop rcm on a computer without GNU Bash installed), but yup, seems good. Will merge soon.

I do have one question about the `[` change, though.

**Comment by fgtham** [comment]
The problem is not with `[`; `dash` doesn't understand the `==` operator:

```
$ dash
> [ "foo" == "foo" ]; echo $?
dash: 3: [: foo: unexpected operator
2
> [ "foo" = "foo" ]; echo $?
0
>
```

So, a fix might be as simple as using `=` as comparison operator. No need for `expr` after all.

**Comment by mike-burns** [comment]
Oh great catch; you're right, `[` does not define `==` but does define `=`. That seems to be the only `==` in the codebase.

Also interesting that Dash uses a built-in `[` instead of the system one. Great for performance.

**Comment by mike-burns** [comment]
Nice, thank you for that last commit. Looks good for me to merge, now I just gotta do it.

Also, thank you thank you for tracking down these bugs!

**Comment by mike-burns** [comment]
Committed as f4f6a0722a3f75b2b1e0d820eb6fc3c9d33fa0eb. Thank you again!

**Pull request closed without merging**